packages/ihe-gateway-sdk/src/common/shared.ts

```diff
@@ -0,0 +1,12 @@
+const OID_REGEX = /^(urn:oid:)?([0-9]+(\.[0-9]+)*)$/;
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions

> Using the constructor function provides runtime compilation of the regular expression. 

```ts
new RegExp("ab+c");
```

Reply:
From the same docs, they're saying to use literal if it the pattern is constant, and constructor if it changes: 

```If the regular expression remains constant, using [regular expression literals] can improve performance.```

```Use the constructor function when you know the regular expression pattern will be changing, or you don't know the pattern and are getting it from another source, such as user input.```

Reply:
Good catch!